### DISCUSSION TITLE
Seems like initConnection defaults to Amazon and never to null

### DISCUSSION THREAD

**Discussion by timbrandin** [hyochan/react-native-iap#2145] [General]
https://github.com/dooboolab/react-native-iap/blob/main/src/internal/platform.ts#L41

We're getting a lot of Errors on BillingResponseCode.FEATURE_NOT_SUPPORTED after calling getSubscriptions since v12.3 of react-native-iap, which could be because of many reasons...

a) Google Play Services is not up to date on your device
b) You are not signed into the device with a google account
c) Your device/Android OS is no longer supported by the new native library
d) You are attempting to sell a subscription in an area that does not support it. We've seen people from India for example report that some subscriptions are no longer supported in their region: subscription upgrade/downgrade not working in android  #1426 https://github.com/dooboolab/react-native-iap/issues/1344
e) Please look at the logs, there might be a warning mentioning that your app isn't calling initConnection which was not enforced in previous versions but now it is

-- 

Our suspicion is against the referenced code, that it defaults to amazon and then initConnection will pass through even though the native android play library is not available.

**Comment by andresesfm** [comment]
@timbrandin The condition you are pointing to checks if the Native Module is present. And the Native module is set at build time, not at runtime. There's no way the amazon implementation leaking into the Android implementation as far as I can see. In order to catch all those edge cases related to the connection, please check the result of the `initConnection` call.
